src/sonic-config-engine/sonic-cfggen

```diff
@@ -328,28 +367,23 @@ def main():
             hardware_data['DEVICE_METADATA']['localhost'].update(asic_id=asic_id)
         deep_update(data, hardware_data)
 
-    if args.template is not None:
-        template_file = os.path.abspath(args.template)
-        paths  = ['/', '/usr/share/sonic/templates', os.path.dirname(template_file)]
-        if args.template_dir is not None:
-            template_dir = os.path.abspath(args.template_dir)
-            paths.append(template_dir)
-        loader = jinja2.FileSystemLoader(paths)
-
-        redis_bcc = RedisBytecodeCache(SonicV2Connector(host='127.0.0.1'))
-        env = jinja2.Environment(loader=loader, trim_blocks=True, bytecode_cache=redis_bcc)
-        env.filters['sort_by_port_index'] = sort_by_port_index
-        env.filters['ipv4'] = is_ipv4
-        env.filters['ipv6'] = is_ipv6
-        env.filters['unique_name'] = unique_name
-        env.filters['pfx_filter'] = pfx_filter
-        env.filters['ip_network'] = ip_network
-        for attr in ['ip', 'network', 'prefixlen', 'netmask', 'broadcast']:
-            env.filters[attr] = partial(prefix_attr, attr)
-        # Pass the is_multi_npu function as global
-        env.globals['multi_asic'] = is_multi_npu
-        template = env.get_template(template_file)
-        print(template.render(sort_data(data)))
+    paths = ['/', '/usr/share/sonic/templates']
+    if args.template_dir:
+        paths.append(os.path.abspath(args.template_dir))
+
+    if args.template:
+        for template_file, _ in args.template:
+            paths.append(os.path.dirname(os.path.abspath(template_file)))
+        env = _get_jinja2_env(paths)
+        sorted_data = sort_data(data)
+        for template_file, dest_file in args.template:
+            template = env.get_template(os.path.basename(template_file))
+            template_data = template.render(sorted_data)
+            if dest_file == "config-db":
```

Comment:
>"config-db" [](start = 28, length = 11)

This is magic strings for a special mode. What if user want to write to a file called 'config-db' ?

Reply:
The user can write and rename the file later. Currently when we save config-db.json, we simply redirect stdout to the file.

Reply:
maybe we can use `-t template-2.j2,[config-db]` to indicate this is config-db, not a config-db file. if user really generate a config-db file, then use `-t template-2.j2,\[config-db\]`

Reply:
the user can use this syntax `-t template-2.j2,/config-db` or `-t template-2.j2,./config-db` to write to file named config-db. The check is strictly done against config-db as keyword. The / would serve as a distinction between the keyword and a filename that happens to be the same as the keyword.

I can changed it to [config-db], however the premise of argument still exist. What if the file name is [config-db]. I tend to think this a user error since there is clear way if config-db is used as a file name.

Reply:
Just tried the ./config-db and it does generate the file as intended:
```
admin@str-s6000-acs-14:~$ sudo sonic-cfggen -d -t /usr/share/sonic/templates/rsyslog.conf.j2,./config-db
admin@str-s6000-acs-14:~$ ls -alrt
total 85
-rw-r--r-- 1 admin admin   807 Apr 18  2019 .profile
-rw-r--r-- 1 admin admin  3526 Apr 18  2019 .bashrc
-rw-r--r-- 1 admin admin   220 Apr 18  2019 .bash_logout
drwxr-xr-x 1 root  root   4096 Aug 11 03:41 ..
drwxr-xr-x 1 admin admin  4096 Aug 12 21:16 .
-rw-r--r-- 1 root  root   1907 Aug 12 21:16 config-db
```